### PR TITLE
Support bullet points in blockquotes

### PR DESCRIPTION
**Pull request by LeMurphant**
fix https://github.com/StampyAI/GDocsRelatedThings/issues/95

I tried to do it in a principled way, but it's not exactly pretty.
image

Also cleaned up the various places that dealt with trailing newlines and centralized them.

Tested it with a bunch of other inputs, did not seem to affect them.

### CODE REVIEW

File: parser/utils.js [hunk 1]
```diff
@@ -4,6 +4,58 @@ import { tableURL, codaColumnIDs, glossaryTableURL } from "./constants.js";
...
+//      - Then any content for that list item (non-greedy)
+//      - Followed by one-or-more blank lines (possibly spaces/tabs)
+//    Lookahead:
+//      - Optional "> " prefixes (quoted), optional indentation
+//      - Another list marker (dash or ordered)
+const RE_BLANKS_BETWEEN_ANY_LIST_ITEMS =
+  /((?:^|\n)(?:>\s*)*\s*(?:-\s|\d+\.\s)[\s\S]*?)\n(?:[ \t]*\n)+(?=(?:>\s*)*\s*(?:-\s|\d+\.\s))/g;
+
+// 5) Extra safety: collapse blanks specifically between two plain dash bullets (non-quoted)
+//    This catches plain lists with an "empty" (whitespace-only) line in between
```
Comment (Aprillion) [inline parser/utils.js:39]: does this "extra" solve any real problem?

File: parser/utils.js [hunk 2]
```diff
@@ -4,6 +4,58 @@ import { tableURL, codaColumnIDs, glossaryTableURL } from "./constants.js";
...
+// Minimal, regex-based spacing normalizer for Markdown
+// Behavior:
+// - Trim trailing spaces per line
+// - Collapse 3+ consecutive newlines to exactly 2
+// - Remove a bare '>' separator right before a quoted list item (keeps quote block tight)
+// - Remove any whitespace-only blank lines between consecutive list items
+//   (works for both quoted and plain lists, and tolerates optional indentation)
+
+// 1) Trailing spaces at EOL
+const RE_TRAILING_SPACES = /[ \t]+$/gm;
```
Comment (Aprillion) [inline parser/utils.js:16]: (minor) do the variables make the code more readable for you? IMHO if everything except RE_BLANKS_BETWEEN_ANY_LIST_ITEMS was inlined without intermediate variables, it would be more readable to me ... (or maybe removing most comments would help to see the definitions without scrolling)

File: parser/utils.js [hunk 3]
```diff
@@ -4,6 +4,58 @@ import { tableURL, codaColumnIDs, glossaryTableURL } from "./constants.js";
 const pprint = (data) => console.log(JSON.stringify(data, null, 2));
 
 // Some answers are HUGE and full of youtube embeds, so we sometimes need to squash them as Coda only lets us push 85KB into their API at a time. 40K of markdown seems to become 95K over the wire, so we set our limit a good few K under that.
+// Minimal, regex-based spacing normalizer for Markdown
```
Comment (Aprillion) [inline parser/utils.js:7]: lol "minimal" .. I haven't checked the updated tests, do we test for all positive&negative cased for each of these regular exceptions (in at least a few combinations)?

File: parser/parser.js [hunk 4]
```diff
@@ -266,10 +296,34 @@ export const mergeSameElements = (elements) =>
     return acc;
   }, []);
 
-const isQuote = (paragraphStyle) => {
+// Determine whether a paragraph should be treated as a Markdown blockquote.
+// Notes about Google Docs indentation units:
+// - A single "indent button" click is ~36pt. Bulleted/numbered lists at nesting level N
+//   typically have a baseline indentation of 36pt per level (i.e. (N+1) * 36pt).
```
Comment (Aprillion) [inline parser/parser.js:302]: "typically" ... none of our docs is atypical? or we don't care if those would render wrongly?

**Review by Aprillion — APPROVED** [review]
nothing in code looks too suspicious if well tested

File: parser/parser.js (hunk 4 above)
Comment (LeMurphant) [reply parser/parser.js:302]: I have not found a better way to detect blockquotes than to test for indentation, which is fickle by nature and can break if somebody uses non-standard indentation. I tested it with a bunch of articles and it works ok, I still think this is the best that can be done right now.

File: parser/utils.js (hunk 3 above)
Comment (LeMurphant) [reply parser/utils.js]: The previous number of tests sat somewhere between -1 and 1
https://github.com/StampyAI/GDocsRelatedThings/pull/96/commits/3d8b18cccbf518de090391280f1d6673c96f0b89 should help
The tests were AI written, but they seem to make sense. I expect that over time we will find edge cases where we will want to fix the tests, but this is probably a local improvement.

**Review by Aprillion — APPROVED** [review]
